pandas/tests/frame/test_constructors.py

```diff
@@ -3052,6 +3053,24 @@ def test_from_dict_with_columns_na_scalar(self):
         expected = DataFrame({"a": Series([pd.NaT, pd.NaT])})
         tm.assert_frame_equal(result, expected)
 
+    # TODO: make this not cast to object in pandas 3.0
+    @pytest.mark.skipif(
+        not np_version_gt2, reason="StringDType only available in numpy 2 and above"
+    )
+    @pytest.mark.parametrize(
+        "data",
+        [
+            {"a": ["a", "b", "c"], "b": [1.0, 2.0, 3.0], "c": ["d", "e", "f"]},
+        ],
+    )
+    def test_np_string_array_object_cast(self, data):
+        from numpy.dtypes import StringDType
+
+        data["a"] = np.array(data["a"], dtype=StringDType())
```

Comment:
Curious what happens when `dtype=StringDType()` is passed to pandas `Series` or `DataFrame` constructor? Fine if it doesn't work/converts to object but would be good to note if something unexpected happens

Reply:
I think it silently casts to object.

FWIW, I think the numpy fixed length strings also do this.